sui_programmability/examples/frenemies/sources/frenemies.move

```diff
@@ -39,8 +39,15 @@ module frenemies::frenemies {
         epoch_score: u16,
     }
 
+    /// Emitted in `confirm_address` call.
+    struct ParticipationProofCommitted has copy, drop {
+        commitment_hash: vector<u8>
```

Comment:
Typically to prove account ownership we sign over a proof of possession challenge (imagine a personal message over some challenge string). 
Other ways are to submit a tx which includes as metadata something that the off chain challenger asked you.

Just committing to a hash onchain,  might not suffice to prove ownership offchain. The reason being you will reveal the pre image to the challenger (off chain verifier), but now this verifier can use your preimage to impersonate you in another challenger. 

TL;DR Typically proof of possession should protect against replay attacks, which means you require some unpredictable non-reusable freshness per challenge (proof), and thus you either do some ZKP or sign over a fresh nonce challenge each time you are requested to prove ownership.

Reply:
Also it’s a last minute change before code goes live.

Reply:
Using hash here to make sure people don’t submit a tx with sensitive data (say email or a discord login).

Reply:
Then don't name it commitment_hash, but rather nonce?